R/geom_flag.R

```diff
@@ -46,34 +57,39 @@ scale_country <- function(..., guide = "legend") {
 GeomFlag <- ggplot2::ggproto("GeomFlag", ggplot2::Geom,
   required_aes = c("x", "y", "country"),
   default_aes = ggplot2::aes(size = 5, country = "nz"),
-  draw_key = function(data, params, size) {
-    flagGrob(0.5, 0.5, country = data$country, size = data$size)
-  },
-  draw_group = function(data, panel_scales, coord) {
+  draw_key = draw_key_flag("nz"),
+  draw_panel = function(data, panel_scales, coord) {
     coords <- coord$transform(data, panel_scales)
-    flagGrob(coords$x, coords$y, coords$country, coords$size)
+
+    make_flag_grob <- function(i) {
+      flagGrob(
+        coords$x[i],
+        coords$y[i],
+        coords$country[i],
+        coords$size[i]
+      )
+    }
+
+    # build a flag for each coord row
+    svg_grobs <- lapply(seq_len(nrow(coords)), make_flag_grob)
+    flag_tree <- do.call(grid::grobTree, svg_grobs)
+    return(flag_tree)
   }
 )
 
 #' @noRd
 flagGrob <- function(x, y, country, size = 1, alpha = 1) {
-  grid::gTree(x = x, y = y, country = country, size = size, cl = "flag")
-}
-
-#' @noRd
-#' @exportS3Method grid::makeContent
-makeContent.flag <- function(x) {
-  flag_pics <- lapply(
-    seq_along(x$country),
-    function(ii) {
-      grImport2::pictureGrob(
-        picture = ggflags::lflags[[x$country[[ii]]]],
-        x = x$x[ii], y = x$y[ii],
-        width = x$size[ii] * grid::unit(1, "mm"),
-        height = x$size[ii] * grid::unit(1, "mm"),
-        distort = FALSE
-      )
-    }
-  )
-  grid::setChildren(x, do.call(grid::gList, flag_pics))
+  flag_size <- size * grid::unit(1, "mm")
+  flag_grob <- ggsvg::svg_to_rasterGrob(
```

Comment:
If I convert the `svg_text` to raw using `charToRaw()`, I can skip `ggsvg` and use `rsvg::rsvg_nativeraster()` instead (which takes the same arguments). That's probably for the best, as I don't think Mike intends for `ggsvg` to be super maintained, and `rsvg` is well maintained.

Reply:
See https://github.com/coolbutuseless/ggsvg/blob/c9d0d490be51602dafcf9fbcaf43be68d5b05085/R/rsvg.R#L17